keps/sig-network/2136-np-version-status/README.md

```diff
@@ -0,0 +1,858 @@
+    policy).
+
+  - If the policy has a recognized `minVersion`, but the plugin can see
+    that it uses features which the plugin does not support, then the
+    plugin should set a status condition with `{ type: "Supported",
+    status: "False", reason: "Unimplemented" }` and an appropriate
+    `message`. It is up to the plugin's discretion whether it should
+    also set a `"False"` `"Enforcing"` status like above, or if it
+    instead chooses to implement as much of the NetworkPolicy as it can.
+
+  - If a policy has a recognized `minVersion` and uses only supported
+    features, the plugin _may_ set a status condition with `{ type:
+    "Supported", status: "True" }`. However, this can also be assumed as
+    the default value if there is no `"Supported"` condition.
+
+Note the distinction between the first two cases; if the `minVersion` is
+higher than the highest version the plugin knows about, then the plugin
+has no way of telling whether it can implement the policy or not, and no
+way of knowing for sure if it's correctly interpreting even the parts
+that it can see. (For example, if a network plugin that only implemented
+version "1.9" received a policy with a combined `podSelector` and
+`namespaceSelector`, it might misinterpret it by reading the
+`namespaceSelector` but not the `podSelector`, and thereby allowing
+access to/from more pods than was intended.) Thus, it is best for it to
+not even try.
+
+On the other hand, if the plugin knows about a feature but doesn't
+implement it, they may still be able to implement the other parts of the
+policy safely. (For example, a plugin that does not implement egress
+policies at all might choose to still implement the ingress side of a
+mixed ingress/egress policy.)
+
+#### The "Enforcing" Condition
+
+The `"Enforcing"` condition indicates whether the plugin is currently
+enforcing the policy described by a NetworkPolicy.
+
+- If the plugin is intentionally not enforcing the policy (eg, because
+  it depends on unsupported features), it should set a `{ type: "Enforcing",
+  status: "False" }` condition, with an appropriate `message` and the
+  same `reason` as the `"Supported"` condition.
+
+- A plugin which is implementing a policy but is not able to determine
+  when the policy is fully in effect should set a status condition with
+  `{ type: "Enforcing", status: "Unknown" }` and an appropriate `reason`
+  and `message`. This indicates to the user that the enforcing status is
+  not known and is not going to become known.
+
+- A plugin which is able to determine when a NetworkPolicy is fully in
+  effect _may_ set an initial status condition with `{ type:
+  "Enforcing", status: "False", reason: "Pending" }`, and an appropriate
+  `message`. In particular, it should do this if it thinks there may be
+  a noticeable delay before it is able to set the condition to `"True"`.
+
+- A plugin which is able to determine when a NetworkPolicy is fully in
```

Comment:
Here there be dragons.  What if a new node joins which isn't enforcing yet?  What if a policy comes undone (e.g. someone flushes iptables).  Especially around security, we need to be careful what we claim to be truth.